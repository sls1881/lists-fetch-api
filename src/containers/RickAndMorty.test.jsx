import React from 'react';
import { render, screen } from '@testing-library/react';
import RickAndMorty from './RickAndMorthy';
import { MemoryRouter } from 'react-router-dom';

describe('Rick and Morty API', () => {
  it('should render Rick and Morty character data', async () => {
    render(
      <MemoryRouter>
        <RickAndMorty />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'characters' });

    expect(ul).toMatchSnapshot();
  });
});
