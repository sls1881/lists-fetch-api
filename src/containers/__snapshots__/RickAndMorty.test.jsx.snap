// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rick and Morty API should render Rick and Morty character data 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      class="links"
      href="/1"
    >
      <div>
        <h2>
          Rick Sanchez
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Rick Sanchez"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/2"
    >
      <div>
        <h2>
          Morty Smith
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Morty Smith"
          src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/3"
    >
      <div>
        <h2>
          Summer Smith
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Summer Smith"
          src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/4"
    >
      <div>
        <h2>
          Beth Smith
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Beth Smith"
          src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/5"
    >
      <div>
        <h2>
          Jerry Smith
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Jerry Smith"
          src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/6"
    >
      <div>
        <h2>
          Abadango Cluster Princess
        </h2>
        <p>
          Alien
        </p>
        <img
          alt="Abadango Cluster Princess"
          src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/7"
    >
      <div>
        <h2>
          Abradolf Lincler
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Abradolf Lincler"
          src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/8"
    >
      <div>
        <h2>
          Adjudicator Rick
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Adjudicator Rick"
          src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/9"
    >
      <div>
        <h2>
          Agency Director
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Agency Director"
          src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/10"
    >
      <div>
        <h2>
          Alan Rails
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Alan Rails"
          src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/11"
    >
      <div>
        <h2>
          Albert Einstein
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Albert Einstein"
          src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/12"
    >
      <div>
        <h2>
          Alexander
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Alexander"
          src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/13"
    >
      <div>
        <h2>
          Alien Googah
        </h2>
        <p>
          Alien
        </p>
        <img
          alt="Alien Googah"
          src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/14"
    >
      <div>
        <h2>
          Alien Morty
        </h2>
        <p>
          Alien
        </p>
        <img
          alt="Alien Morty"
          src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/15"
    >
      <div>
        <h2>
          Alien Rick
        </h2>
        <p>
          Alien
        </p>
        <img
          alt="Alien Rick"
          src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/16"
    >
      <div>
        <h2>
          Amish Cyborg
        </h2>
        <p>
          Alien
        </p>
        <img
          alt="Amish Cyborg"
          src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/17"
    >
      <div>
        <h2>
          Annie
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Annie"
          src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/18"
    >
      <div>
        <h2>
          Antenna Morty
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Antenna Morty"
          src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/19"
    >
      <div>
        <h2>
          Antenna Rick
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Antenna Rick"
          src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
        />
      </div>
    </a>
  </li>
  <li>
    <a
      class="links"
      href="/20"
    >
      <div>
        <h2>
          Ants in my Eyes Johnson
        </h2>
        <p>
          Human
        </p>
        <img
          alt="Ants in my Eyes Johnson"
          src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
        />
      </div>
    </a>
  </li>
</ul>
`;
